spec/models/miq_report/generator_spec.rb

```diff
@@ -0,0 +1,118 @@
+          :name            => "Host Memory Utilization Trends (last week)",
+          :title           => "TEST Custom VM report",
+          :rpt_group       => "Custom",
+          :rpt_type        => "Custom",
+          :db              => "VimPerformanceTrend",
+          :order           => "Ascending",
+          :sortby          => ["resource_name"],
+          :time_profile_id => @time_profile_exl.id,
+          :db_options      => {:limit_col    => "max_derived_memory_available",
+                               :trend_col    => "derived_memory_used",
+                               :rpt_type     => "trend",
+                               :start_offset => 604_800,
+                               :end_offset   => 0,
+                               :trend_db     => "HostPerformance",
+                               :interval     => "daily",
+                               :target_pcts  => [70, 80, 90]})
+
+        t = Time.current
+        day_midnight = Time.new(t.year, t.month, t.day, 0, 0, 0, "+00:00").utc - 1.day
+        @used_mem_up = [700, 600, 500, 400]
+        @used_mem_up.size.times do |i|
+          @host1.metric_rollups << FactoryGirl.create(:metric_rollup_host_daily,
+                                                      :timestamp                => day_midnight - i.day,
+                                                      :time_profile_id          => @time_profile_all.id,
+                                                      :derived_memory_used      => @used_mem_up[i],
+                                                      :derived_memory_available => 1400)
+        end
+        used_mem_down = [300, 400, 500, 600]
+        used_mem_down.size.times do |i|
+          @host2.metric_rollups <<  FactoryGirl.create(:metric_rollup_host_daily,
+                                                       :timestamp                => day_midnight - i.day,
+                                                       :time_profile_id          => @time_profile_all.id,
+                                                       :derived_memory_used      => used_mem_down[i],
+                                                       :derived_memory_available => 1400)
+        end
+        used_mem_flat = [301, 300, 300, 301]
+        used_mem_flat.size.times do |i|
+          @host1.metric_rollups <<  FactoryGirl.create(:metric_rollup_host_daily,
+                                                       :timestamp                => day_midnight - i.day,
+                                                       :time_profile_id          => @time_profile_exl.id,
+                                                       :derived_memory_used      => used_mem_flat[i],
+                                                       :derived_memory_available => 1400)
+        end
+
+        @miq_report_profile_all.generate_table(:userid => @user.userid)
+        @miq_report_profile_exl.generate_table(:userid => @user.userid)
```

Comment:
The setup seems pretty (perhaps unavoidably) heavy, and resembles a fixture style of testing. I'm wondering if this could benefit from moving objects that we're creating for every test such as `@miq_report_profile_exl` into the only test that needs it, or saving that into a separate context for the only tests that need it/them.